### PR TITLE
Search navigation pt 1.

### PR DESCRIPTION
**Pull request by mdbenjam**
Connects: #3659 

This is the first part of navigating search results. Now you can hit enter/shift + enter to increment/decrement the index of the current result. Next we can hook this up to the highlighted text.

**Test Plan**
- [ ] Search for a term and hit enter/shift + enter to see the index go up and down mod the total count.

### CODE REVIEW

File: client/app/reader/selectors.js [hunk 1]
```diff
@@ -107,3 +107,10 @@ export const getTotalMatchesInFile = createSelector(
     map((match) => match.matches).
     sum()
 );
+
+const getSelectedIndex = (state) => state.readerReducer.documentSearchIndex;
+
+export const getCurrentMatchIndex = createSelector(
```
Comment (mdbenjam) [inline client/app/reader/selectors.js]: I was going to put this in the state, but I needed the getTotalMatchesInFile function which was only available in the selectors.

File: client/app/reader/selectors.js [hunk 2]
```diff
@@ -96,7 +96,7 @@ export const getMatchesPerPageInFile = createSelector(
   (matchedPages, txt) => _(matchedPages).
     map((page) => ({
       id: page.id,
-      matches: page.text.match(new RegExp(txt, 'gi')).length
+      matches: (page.text.match(new RegExp(txt, 'gi')) || []).length
```
Comment (mdbenjam) [inline client/app/reader/selectors.js]: Currently fails if there are no matches.

Comment (dannysellers) [reply client/app/reader/selectors.js]: good call 👍

**Comment by mdbenjam** [comment]
Some great points! When Gina has the designs ready, we can reevaluate how people will navigate.

**Pull request merged**